internal/blocktx/processor.go

```diff
@@ -707,20 +730,30 @@ func (p *Processor) storeTransactions(ctx context.Context, blockID uint64, block
 			break
 		}
 
-		bump, err := bc.NewBUMPFromMerkleTreeAndIndex(block.Height, merkleTree, uint64(txIndex)) // #nosec G115
-		if err != nil {
-			return errors.Join(ErrFailedToCreateBUMP, fmt.Errorf("tx hash %s, block height: %d", hash.String(), block.Height), err)
+		tx := store.TxWithMerklePath{
+			Hash:            hash[:],
+			MerklePath:      "",
+			MerkleTreeIndex: int64(txIndex), // #nosec G115
 		}
 
-		bumpHex, err := bump.String()
-		if err != nil {
-			return errors.Join(ErrFailedToGetStringFromBUMPHex, err)
+		_, found := registeredTxsMap[string(hash[:])]
+
+		// only calculate merklePath for transactions that are registered
+		if found {
+			bump, err := bc.NewBUMPFromMerkleTreeAndIndex(block.Height, merkleTree, uint64(tx.MerkleTreeIndex)) // #nosec G115
```

Comment:
I don't really like unhandled errors. I know that this will only return an error when the merkle tree is empty, but this is an external lib, so what if the implementation changes? Or what if we accidentally pass an empty merkle tree there?

Anyways, in this case, we will have a panic earlier anyways, because we're never checking the length of merkle tree, so I removed handling the error here.